src/main/java/mServer/crawler/sender/MediathekZdf.java

```diff
@@ -34,208 +34,221 @@
+        Log.sysLog("convert VideoDTO to DatenFilm finished.");
+      }
     }
 
-    private static void updateHdStatus(DatenFilm film, String urlSeite) {
-        // manuell die Auflösung für HD setzen, 2 Versuche
-        updateHd("1456k_p13v12.mp4", "3328k_p36v12.mp4", film, urlSeite);
-        updateHd("2256k_p14v12.mp4", "3328k_p36v12.mp4", film, urlSeite);
-        updateHd("2328k_p35v12.mp4", "3328k_p36v12.mp4", film, urlSeite);
+    perfPoint.collect();
+  }
 
-        updateHd("1456k_p13v12.mp4", "3256k_p15v12.mp4", film, urlSeite);
-        updateHd("2256k_p14v12.mp4", "3256k_p15v12.mp4", film, urlSeite);
-        updateHd("2328k_p35v12.mp4", "3256k_p15v12.mp4", film, urlSeite);
+  void shutdownAndAwaitTermination(ExecutorService pool, long delay, TimeUnit delayUnit) {
+    pool.shutdown();
+    try {
+      if (!pool.awaitTermination(delay, delayUnit)) {
+        pool.shutdownNow();
+        if (!pool.awaitTermination(delay, delayUnit)) {
+          Log.sysLog("Pool did not terminate");
+        }
+      }
+    } catch (InterruptedException ie) {
+      pool.shutdownNow();
+      Thread.currentThread().interrupt();
+    }
+  }
 
-        updateHd("1496k_p13v13.mp4", "3296k_p15v13.mp4", film, urlSeite);
-        updateHd("2296k_p14v13.mp4", "3296k_p15v13.mp4", film, urlSeite);
-        updateHd("2328k_p35v13.mp4", "3296k_p15v13.mp4", film, urlSeite);
+  @SuppressWarnings("serial")
+  private class VideoDtoDatenFilmConverterAction extends RecursiveAction {
 
-        updateHd("1496k_p13v13.mp4", "3328k_p36v13.mp4", film, urlSeite);
-        updateHd("2296k_p14v13.mp4", "3328k_p36v13.mp4", film, urlSeite);
-        updateHd("2328k_p35v13.mp4", "3328k_p36v13.mp4", film, urlSeite);
-    }
+    private final VideoDTO video;
 
-    private static void modifyUrl(DatenFilm film, String urlSeite, FilmeSuchen mSFilmeSuchen) {
-        //große URL verbessern
-        changeUrl("2256k_p14v11.mp4", "2328k_p35v11.mp4", film, urlSeite, mSFilmeSuchen);
-        changeUrl("2256k_p14v12.mp4", "2328k_p35v12.mp4", film, urlSeite, mSFilmeSuchen);
-        changeUrl("2296k_p14v13.mp4", "2328k_p35v13.mp4", film, urlSeite, mSFilmeSuchen);
+    public VideoDtoDatenFilmConverterAction(VideoDTO aVideoDTO) {
+      video = aVideoDTO;
+      phaser.register();
+    }
 
-        //klein nach groß
-        changeUrl("1456k_p13v11.mp4", "2328k_p35v11.mp4", film, urlSeite, mSFilmeSuchen);
-        changeUrl("1456k_p13v11.mp4", "2256k_p14v11.mp4", film, urlSeite, mSFilmeSuchen); //wenns nicht geht, dann vielleicht so
+    @Override
+    protected void compute() {
+      if (video != null) {
+        try {
+          DownloadDTO download = video.getDownloadDto();
 
-        changeUrl("1456k_p13v12.mp4", "2328k_p35v12.mp4", film, urlSeite, mSFilmeSuchen);
-        changeUrl("1456k_p13v12.mp4", "2256k_p14v12.mp4", film, urlSeite, mSFilmeSuchen); //wenns nicht geht, dann vielleicht so
+          download.getLanguages().forEach(language -> addFilm(download, language));
```

Comment:
bei nur wenigen Sprachen die auftreten, sollte kein Stream verwendet werden, da das unnötiger Overhead ist, der dann auch sehr langsam ist. Siehe https://jaxenter.com/java-performance-tutorial-how-fast-are-the-java-8-streams-118830.html bzw. die Beispiele in einem alten Issue https://github.com/mediathekview/MServer/issues/68#issuecomment-278739894

Reply:
Danke für den Hinweis.